### DISCUSSION TITLE
Unable to download Magisk via wsl

### DISCUSSION THREAD

**Discussion by Sky076** [LSPosed/MagiskOnWSALocal#330] [Q&A] [answered]
The magisk can be downloaded directly from browser with the link but get error in wsl. 
Even if I put the magisk file into the download folder, it still keeps error, no matter what file name I use. 

2023-01-07 17:18:42.729089 [ERROR] [AbstractCommand.cc:349] CUID#10 - Download aborted. URI=https://cdn.jsdelivr.net/gh/topjohnwu/magisk-files@5b2934603f979d7133e67bbc30c6ce3fa0d8e31e/app-release.apk
Exception: [AbstractCommand.cc:351] errorCode=19 URI=https://raw.githubusercontent.com/topjohnwu/magisk-files/5b2934603f979d7133e67bbc30c6ce3fa0d8e31e/app-release.apk
  -> [NameResolver.cc:58] errorCode=19 Failed to resolve the hostname raw.githubusercontent.com, cause: No address associated with hostname

 ...

2023-01-07 17:18:43.560467 [NOTICE] [RequestGroupMan.cc:424] Download GID#bb0f7bcdfa2b1473 not complete: /home/sky/MagiskOnWSALocal/download/magisk-canary.zip
2023-01-07 17:18:43.921327 [INFO] [BackupIPv4ConnectCommand.cc:85] CUID#15 - Backup connection canceled

![image](https://user-images.githubusercontent.com/97520054/211143973-9da9307c-e7ba-4553-a6b5-57f394ef1657.png)

**Comment by Howard20181** [comment, marked as answer]
Check your DNS setting (maybe you should setup proxy in WSL) or use offline mode

**Reply by 741292873** [reply]
how to use offline mode? Thx.

**Comment by YusufYuso** [comment]
see this link : https://cdn.jsdelivr.net/gh/topjohnwu/magisk-files@5b2934603f979d7133e67bbc30c6ce3fa0d8e31e/app-release.apk 
try to download app-release.apk
if it doesn't downloaded and got an issue then try using vpn usa
after you got the apk 
put it in your download folder and change it's name to magisk-canary.zip 
and finally do using `./build.sh --magisk-custom` to skip the download
it worked for me.
